torchgeo/trainers/byol.py

```diff
@@ -368,11 +361,13 @@ def __init__(self, **kwargs: Any) -> None:
             ValueError: if kwargs arguments are invalid
         """
         super().__init__()
-        self.save_hyperparameters()  # creates `self.hparams` from kwargs
+
+        # Creates `self.hparams` from kwargs
+        self.save_hyperparameters()  # type: ignore[operator]
 
         self.config_task()
 
-    def forward(self, x: Tensor) -> Any:  # type: ignore[override]
+    def forward(self, *args: Any, **kwargs: Any) -> Any:
         """Forward pass of the model.
 
         Args:
```

Comment:
Similar to `training_step`, docstring doesn't match

Reply:
I think the docstring for https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#forward is pretty bad. `*args` and `**kwargs` aren't arguments, they are generic placeholders for arguments and keyword arguments. So for a function with this signature, if we accept arguments like `x` and `y` and keyworks arguments like `compute_foo=True` the proper docstring would be:
```rst
Args:
    x: ...
    y: ...

Keyword Args:
    compute_foo: ...
```
Not:
```rst
Args:
    *args: ...
    **kwargs: ...
```
The latter simply isn't useful since it doesn't tell you _what_ args and kwargs.